src/main/java/com/zendesk/maxwell/util/ListWithDiskBuffer.java

```diff
@@ -51,6 +51,7 @@ public boolean isEmpty() {
 	public T getLast() {
 		return list.getLast();
 	}
+	public T getFirst() { return list.getFirst(); }
```

Comment:
I don't think this is correct in the case where the list has overflowed onto disk - can you add a test to confirm? If it's expensive to grab the element from disk, we could potentially work around the need for this by capturing the first element's timestamp when we add it, rather than at the end of the transaction.

Reply:
Good point. Will do that.